src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -1087,6 +1091,101 @@ def aks_get_credentials(cmd, client, resource_group_name, name, admin=False,
 }
 
 
+# pylint: disable=line-too-long
+def aks_diag(cmd, client, resource_group_name, name, storage_account_id=None):
+    mc = client.get(resource_group_name, name)
+
+    yaml_files = urlopen("https://raw.githubusercontent.com/Azure/aks-diagnostic-tool/master/deployment/aks-diagnostic.yaml")
+    after_split = yaml_files.read().decode().split("---")
+    ds_yaml = after_split[0]
+    secret_yaml = after_split[1]
+    containerlogs_cm = after_split[2]
+    kubeobjects_cm = after_split[3]
+
+    if storage_account_id is None:
+        print("No storage account specified. Try getting storage account from diag settings")
+        storage_account_id = get_storage_account_from_diag_settings(cmd.cli_ctx, resource_group_name, name)
+        if storage_account_id is None:
+            raise CLIError("Must specify a stroage account if no one is setup in diag settings.")
```

Comment:
Maybe it should say, "A storage account must be specified, since there isn't one in the diag settings."